lib/nonnative/grpc_server.rb

```diff
@@ -2,13 +2,14 @@
 
 module Nonnative
   class GRPCServer < Nonnative::Server
-    def initialize(port)
+    def initialize(port, timeout)
       @server = GRPC::RpcServer.new
 
       server.add_http2_port("0.0.0.0:#{port}", :this_port_is_insecure)
+      server.wait_till_running(timeout)
```

Comment:
@tedor - From the docs we see `If run has not been called, this returns immediately.`, so this change has no effect. We are calling `run_till_terminated` which I don't think we need to as we can call `run`.

Could you be so kind to please raise an issue with what you are experiencing and we can then work on a fix 😄

Reply:
The starting happens in it's own thread so this will likely not help.

Reply:
I have changed it to use run https://github.com/alexfalkowski/nonnative/pull/30

Reply:
I see your point thanks. 

Locally I tried to fix it adding these lines of code in the configuration:
```
  class Server < Nonnative::GRPCServer
    # @param grpc [GRPC::RpcServer]
    def configure(grpc)
      grpc.wait_till_running(10)
      grpc.handle(Product::Service::Info.new)
    end
  end
```

But as it listens to separate thread, most likely it works as a simple timeout, which "fix" the server issue for me. And definitely it doesn't make any sense.

Thanks for your fix, let me try how `run` command works.

Reply:
I don't think the `run` command will fix this. So we might need to add the `wait_till_running ` somewhere. Let me think this through.